Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs

```diff
@@ -11,10 +11,10 @@ internal class ObservableItemsSource : IItemsViewSource
 		readonly UICollectionView _collectionView;
 		readonly IList _itemsSource;
 
-		public ObservableItemsSource(IEnumerable itemSource, UICollectionView collectionView)
+		public ObservableItemsSource(IList itemSource, UICollectionView collectionView)
```

Comment:
Why was this changed to IList?

Reply:
This was changed to `IList` because this source only understands lists, not any arb collection type. For example:
`public object this[int index]` and `public int Count`. What was happening was the constructor was just doing a straight cast, but this could cause issues in the future. Since this was just for observable collections, they are `IList`. In addition the factory has it defined as: `case IList _ when itemsSource is INotifyCollectionChanged:` - basically `IList` again.

I foresee evil where Bob comes along and says: "ooh, look. I can wrap my queue with this and magic will happen." and the only magic is a major 💥

Reply:
Well, it's `internal`, so Bob would have to slip it by us, but okay. I'm cool with changing it to `IList`, but for consistency it should also be changed in the Android version.